### ISSUE TITLE
command complexity / Big-O

### ISSUE THREAD

**Issue by mrkamel** [yinqiwen/ardb#394]
Hi and thanks for ardb!

It would be super-helpful to have some doc listing the complexity of the commands, optimally using Big-O notation, like redis does in its docs.

**Comment by yinqiwen** [comment]
the storage engine store data in disk, and there is cache for some operations, it's hard to estimate the   complexity.  
and since rocksdb/leveldb use tree like structure to store data, the basic search complexity  would be log2(N).

**Comment by shailesh33** [comment]
+1 on the request. Essentially what I am looking for is, if I am doing a
HSET, how many reads and writes does it translate to. Similarly for sorted
set operations etc.

**Comment by mrkamel** [comment]
yes, let's forget about the storage layer complexity. I was as well only talking about the complexity that is ardb specific, because even from looking into the ardb code it's not always easy to estimate it, even though it is well written - and btw, who wants to always look into the code to estimate the complexity of ZADD, HSET or SREM.

**Comment by shailesh33** [comment]
@yinqiwen Appreciate your response here. thanks.
